thenewboston_node/business_logic/blockchain/base/network.py

```diff
@@ -67,3 +68,27 @@ def get_primary_validator(self, block_number: Optional[int] = None) -> Optional[
                 return self.get_node_by_identifier(account_number)
 
         return None
+
+    def get_node_role(self, identifier: hexstr) -> Optional[NodeRole]:
+        last_block_number = self.get_last_block_number()
+        node = self.get_node_by_identifier(identifier, last_block_number)
+
+        if node is None:
+            logger.warning(f'Getting node role of non-existent node: {identifier}')
+            return None
+
+        schedule = self.get_account_state_attribute_value(identifier, 'primary_validator_schedule', last_block_number)
```

Comment:
maybe we need to have `get_primary_validator_schedule()` method?

Reply:
@dmugtasimov we alreasdy have `get_primary_validator_schedule` in `account_state.py`
but I used `get_account_state_attribute_value` as `get_primary_validator_schedule` is not in `base.py` module and lint fails.
Should i move `get_primary_validator_schedule` to `base.py` ?

Reply:
@fonar just leave it as is, I will take a look myself after merge

Reply:
fixed here: https://github.com/thenewboston-developers/thenewboston-node/pull/338